### PR TITLE
ACL nits.

### PR DESCRIPTION
**Pull request by srinibas-zededa**
some corner cases for ACLs.
1.  For failed ACL updates, for an active or deactivated(config) app instance, should not kill/release resources for the app instance in run state.  The app instance will keep running or, remain in deactivated state, with the last good state. Added checks for preventing, release ACL resources, for an active/deactivated app instances , with update error.
2. While releasing ACLs for app instances in error state, the ACL lists were not getting properly updated.

### CODE REVIEW

**Comment by rvs** [comment]
Hey @srinibas-zededa can you please update the description for this pull request? It is not clear what is your objective here. Please provide [somewhat] detailed explanation.

**Comment by srinibas-zededa** [comment]
> Hey @srinibas-zededa can you please update the description for this pull request? It is not clear what is your objective here. Please provide [somewhat] detailed explanation.
Added the desciption statement, thanks for prompting me to explain the issue and fix.

File: pkg/pillar/cmd/zedrouter/zedrouter.go [hunk 1]
```diff
@@ -2812,23 +2812,23 @@ func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext, key0 string) {
...
-	for _, olStatus := range status.OverlayNetworkList {
+	for idx, olStatus := range status.OverlayNetworkList {
 		aclArgs := types.AppNetworkACLArgs{BridgeName: olStatus.Bridge,
 			VifName: olStatus.Vif}
 		ruleList, err := deleteACLConfiglet(aclArgs, olStatus.ACLRules)
 		if err != nil {
 			addError(ctx, status, "deleteACL", err)
 		}
-		olStatus.ACLRules = ruleList
+		status.OverlayNetworkList[idx].ACLRules = ruleList
```
[the remark this answers is not in the document]
Comment (srinibas-zededa) [reply pkg/pillar/cmd/zedrouter/zedrouter.go]: olStatus is a local scope reference by value variable, and the changes in this variable, is not  getting written to the status variable. Hence, while publishing the status through pubsub, the intended changes are not there, resulting in failures like double delete/add for the same ACLs.

File: pkg/pillar/cmd/zedrouter/zedrouter.go [hunk 2]
```diff
@@ -2812,23 +2812,23 @@ func scanAppNetworkStatusInErrorAndUpdate(ctx *zedrouterContext, key0 string) {
...
-	for _, ulStatus := range status.UnderlayNetworkList {
+	for idx, ulStatus := range status.UnderlayNetworkList {
 		aclArgs := types.AppNetworkACLArgs{BridgeName: ulStatus.Bridge,
 			VifName: ulStatus.Vif}
 		ruleList, err := deleteACLConfiglet(aclArgs, ulStatus.ACLRules)
 		if err != nil {
 			addError(ctx, status, "deleteACL", err)
 		}
-		ulStatus.ACLRules = ruleList
+		status.UnderlayNetworkList[idx].ACLRules = ruleList
```
[the remark this answers is not in the document]
Comment (srinibas-zededa) [reply pkg/pillar/cmd/zedrouter/zedrouter.go]: Same as above.

File: pkg/pillar/cmd/zedrouter/zedrouter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rvs) [reply pkg/pillar/cmd/zedrouter/zedrouter.go]: That makes perfect sense -- got confused there for a moment.

**Pull request merged**